File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/decryption/Presenter.java
package gateway.hitrontech.com.encryption.fragment.decryption;

import com.hitrontech.hitronencryption.EncryptionManager;
import gateway.hitrontech.com.encryption.utils.Constants;
import gateway.hitrontech.com.encryption.utils.SharePreManager;

public class Presenter implements Contract.Presenter {

  private Contract.View mView;

  public Presenter(Contract.View view) {
    mView = view;
    mView.setPresenter(this);
  }

  @Override
  public void getPlainText(String cipherText) {
    mView.setPlainText(
        EncryptionManager.getInstance()
            .base64DecoderByAppId(
                SharePreManager.getInstance().getAppId(),
                Constants.KEY,
                cipherText
            ));
  }

  @Override
  public void getCipherText() {
    mView.setCipherText(SharePreManager.getInstance().getCipherText());
  }

  @Override
  public void start() {
    getCipherText();
  }
}

File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/decryption/Presenter.kt
package gateway.hitrontech.com.encryption.fragment.decryption

import com.hitrontech.hitronencryption.EncryptionManager
import gateway.hitrontech.com.encryption.utils.Constants
import gateway.hitrontech.com.encryption.utils.SharePreManager

class Presenter(private val mView: Contract.View) : Contract.Presenter {

    init {
        mView.setPresenter(this)
    }

    override fun getPlainText(cipherText: String) {
        mView.showProgressDialog()
        mView.setPlainText(
                EncryptionManager.instance
                        .base64DecoderByAppId(
                                SharePreManager.instance.appId,
                                Constants.KEY,
                                cipherText
                        )!!)
        mView.dismissProgressDialog()
    }

    override fun getCipherText() {
        mView.showProgressDialog()
        mView.setCipherText(SharePreManager.instance.cipherText!!)
        mView.dismissProgressDialog()
    }

    override fun start() {
        getCipherText()
    }
}
